packages/cyberstorm/src/components/Breadcrumb.tsx

```diff
@@ -0,0 +1,86 @@
+import { Fragment, FunctionComponent, ReactNode } from "react";
+
+import { faHouse, faSlashForward } from "@fortawesome/pro-regular-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import styles from "./componentStyles/Breadcrumb.module.css";
+import { IndexLink } from "./Links";
+
+type Label = ReactNode | string;
+
+interface Crumb {
+  label: Label;
+  LinkComponent?: FunctionComponent;
+}
+
+type LinkCrumb<P> = Omit<Crumb, "LinkComponent"> & {
+  label: Label;
+  LinkComponent: FunctionComponent<P>;
+  LinkProps: P;
+};
+
+interface BreadCrumbProps<P> {
+  parts: (Crumb | LinkCrumb<P>)[];
+}
+
+export function BreadCrumb<P>(props: BreadCrumbProps<P>) {
+  if (!props.parts) {
+    return null;
+  }
+  const parts = [...props.parts];
+  const leaf = parts.pop();
+
+  return (
+    <div className={styles.root}>
+      {parts.map((crumb, i) => (
+        <Fragment key={`${crumb.label}-${i}`}>
+          <MidCrumb crumb={crumb} />
+          <Separator />
+        </Fragment>
+      ))}
+
+      <LeafCrumb crumb={leaf} />
+    </div>
+  );
+}
+
+function MidCrumb<P>(props: { crumb: Crumb | LinkCrumb<P> }) {
+  const { crumb } = props;
+  if (crumb.LinkComponent === undefined) {
+    return CrumbLabel(crumb.label);
+  }
+  if (!("LinkProps" in crumb)) {
+    return <crumb.LinkComponent>{CrumbLabel(crumb.label)}</crumb.LinkComponent>;
+  }
+  return (
+    <crumb.LinkComponent {...crumb.LinkProps}>
+      {CrumbLabel(crumb.label)}
+    </crumb.LinkComponent>
+  );
+}
+
+function LeafCrumb<P>(props: { crumb?: Crumb | LinkCrumb<P> }) {
+  const { crumb } = props;
+  if (crumb === undefined) {
+    return null;
+  } else {
+    return CrumbLabel(crumb.label);
+  }
+}
+
+function CrumbLabel(label: Label) {
+  if (label && typeof label === "string") {
+    return <p className={styles.label}>{label}</p>;
```

Comment:
@gkurck this should probably be a span right?

Reply:
The last one without an anchor should be a span, yes. Those inside anchors probably don't need to be additionally wrapped in anything other than the anchor.